src/dotnet-install.sh

```diff
@@ -921,25 +921,24 @@ get_http_header_wget() {
     local remote_path="$1"
     local disable_feed_credential="$2"
     local wget_options="-q -S --spider --tries 5 "
-    # Store options that aren't supported on all wget implementations separately.
-    local wget_options_extra="--waitretry 2 --connect-timeout 15 "
-    local wget_result=''
+
+    local wget_options_extra=''
+
+    # Test for options that aren't supported on all wget implementations.
+    if ! wget -h 2>&1 | grep "waitretry" >/dev/null && wget -h 2>&1 | grep "connect-timeout" >/dev/null; then
+        say 'wget extra options are unavailable for this environment'
+    else
+        wget_options_extra="--waitretry 2 --connect-timeout 15 "
+    fi
 
     remote_path_with_credential="$remote_path"
     if [ "$disable_feed_credential" = false ]; then
         remote_path_with_credential+="$feed_credential"
     fi
 
     wget $wget_options $wget_options_extra "$remote_path_with_credential" 2>&1
```

Comment:
Since we are getting rid of usages of `$?`, I think it may be better to do:
```suggestion
    return wget $wget_options $wget_options_extra "$remote_path_with_credential" 2>&1
```

Reply:
it produces an error
![image](https://user-images.githubusercontent.com/95473390/157055594-c1cc55cb-9b42-421e-9533-72120611db8c.png)

Reply:
Does this work with `sudo` without the suggested fix? If yes, let's keep as is.